ansible/roles/test/files/ptftests/fib_test.py

```diff
@@ -479,26 +481,51 @@ def check_hit_count_map(self, dest_port_list, port_hit_cnt):
         logging.info("%-10s \t %-10s \t %10s \t %10s \t %10s" % ("type", "port(s)", "exp_cnt", "act_cnt", "diff(%)"))
         result = True
 
-        total_hit_cnt = 0
-        for ecmp_entry in dest_port_list:
-            for member in ecmp_entry:
-                total_hit_cnt += port_hit_cnt.get(member, 0)
-
-        for ecmp_entry in dest_port_list:
-            total_entry_hit_cnt = 0
-            for member in ecmp_entry:
-                total_entry_hit_cnt += port_hit_cnt.get(member, 0)
-            (p, r) = self.check_within_expected_range(total_entry_hit_cnt, float(total_hit_cnt)/len(dest_port_list))
-            logging.info("%-10s \t %-10s \t %10d \t %10d \t %10s"
-                         % ("ECMP", str(ecmp_entry), total_hit_cnt//len(dest_port_list), total_entry_hit_cnt, str(round(p, 4)*100) + '%'))
-            result &= r
-            if len(ecmp_entry) == 1 or total_entry_hit_cnt == 0:
-                continue
-            for member in ecmp_entry:
-                (p, r) = self.check_within_expected_range(port_hit_cnt.get(member, 0), float(total_entry_hit_cnt)/len(ecmp_entry))
+        asic_list = defaultdict(list)
+        if self.switch_type == "voq":
+            asic_list['voq'] = dest_port_list
+        else:
+            for port in dest_port_list:
+                if type(port) == list:
+                    port_map = self.ptf_test_port_map[str(port[0])]
+                    asic_id = port_map.get('asic_idx',0)
+                    member = asic_list.get(asic_id)
+                    if member is None:
+                        member = [];member.append(port)
+                    else:
+                        member.append(port)
+                    asic_list[asic_id] = member
+                else:
+                    port_map = self.ptf_test_port_map[str(port)]
+                    asic_id = port_map.get('asic_idx',0)
+                    member = asic_list.get(asic_id)
+                    if member is None:
+                        member = [];member.append(port)
+                    else:
+                        member.append(port)
+                    asic_list[asic_id] = member
+
+        for asic_member in asic_list.values():
+            total_hit_cnt = 0
```

Comment:
As I understand it, we are checking the for a given asic we are load balancing properly on each ecmp path. Do we need to check if the total packets are load balanced across all the asic.

Reply:
Hi @arlakshm, In my opinion you don't need to look at the total packets, because if there is a deviation (due to a bug) which causes one asic to get more packet then the other, expected count will catch the deviation in the same way that it currently does for per destination port level

Reply:
Hi @arlakshm and @abdosi, Thanks for catching the case wherein what if in the first hop the load balancing gets skewed due to a bug. I have made the change and now the expected hit count will be first calculated based on number of asics and then based on the number of outgoing interfaces per asic. Kindly take a look and let me know.